### ISSUE TITLE
Allow collapsing of a machine / state / parallel when clicking it

### ISSUE THREAD

**Issue by mdpratt** [statelyai/xstate-viz#281]
As a user, it would be great if we could click on a large machine / state / parallel and have it collapse, hiding it's rendering.

Bonus points if this preference could be saved to localStorage so that we don't have to collapse our selection on rerenders.

**Comment by brianjuhl** [comment]
Yeah this would be awesome. I also liked how in the old visualizer parallel states were displayed horizontally as instead of vertically.
